examples/knative/powercli/kn-pcli-pg-check/handler.ps1

```diff
@@ -0,0 +1,319 @@
+                     }
+                  }
+               }
+
+               # If none of the portgroup's tags are a match, the vNIC is added to the hash table
+               if ($pgMatch -eq $false ) {
+                  Write-Host "$(Get-Date) - INFO: No permitted tags were found on the portgroup`n"
+                  $networkAdapterHash[$nic.id] = "$($pg.Name)"
+               }
+            }
+         }
+
+         # Check to see if the hash is empty
+         if ( $networkAdapterHash.Count -gt 0 ) {
+            $msg = "NICs using unapproved portgroups:`n"
+            Write-Host "$(Get-Date) - $($msg)"
+            # Build a list of NICs and unapproved portgroups
+            foreach ($nic in $networkAdapters) {
+               if ($networkAdapterHash.ContainsKey($nic.Id)) {
+                  Write-Host $nic.Name, $networkAdapterHash[$nic.Id]
+                  $msg += $nic.Name + " - " + $networkAdapterHash[$nic.Id] + "`n"
+               }
+            }
+
+            # Payload for Slack
+            $payload = @{
+               attachments = @(
+                  @{
+                     pretext = $(${jsonSecrets}.SLACK_MESSAGE_PRETEXT);
+                     fields = @(
+                        @{
+                              title = "EventType";
+                              value = $cloudEvent.Subject;
+                              short = "false";
+                        }
+                        @{
+                              title = "Username";
+                              value = $cloudEventData.UserName;
+                              short = "false";
+                        }
+                        @{
+                              title = "DateTime";
+                              value = $cloudEventData.CreatedTime;
+                              short = "false";
+                        }
+                        @{
+                           title = "Full Message";
+                           value = $msg + "`n`n" + $cloudEventData.FullFormattedMessage ;
+                           short = "false";
+                        }
+                     )
+                  }
+               )
+            }
+
+            # Convert Slack message object into JSON
+            $body = $payload | ConvertTo-Json -Depth 5
+
+            if(${env:FUNCTION_DEBUG} -eq "true") {
+               Write-Host "$(Get-Date) - DEBUG: `"$body`""
+            }
+
+            Write-Host "$(Get-Date) - Sending Webhook payload to Slack ..."
+            $ProgressPreference = "SilentlyContinue"
+
+            try {
+               Invoke-WebRequest -Uri $(${jsonSecrets}.SLACK_WEBHOOK_URL) -Method POST -ContentType "application/json" -Body $body
+            } catch {
+               throw "$(Get-Date) - Failed to send Slack Message: $($_)"
+            }
+
+            Write-Host "$(Get-Date) - Successfully sent Webhook ..."
+          } else {
+            Write-Host "$(Get-Date) - INFO: All NICs are on approved portgroups"
```

Comment:
nit: inverse logic to reduce if/else nesting

Reply:
Fixed